xuemingjun/server/src/server_node.cpp

```diff
@@ -0,0 +1,43 @@
+//
+// Created by mijiao on 23-1-28.
+//
+#include <client/message.h>
+#include <server/service.h>
+#include <ros/ros.h>
+#include <string>
+#include <map>
+
+
+std::map<std::string, ros::Subscriber> sublist;
+ros::NodeHandle *nhp = nullptr;
+
+void topicCallBack(const client::message::ConstPtr &msg) {
+    ROS_INFO("From %s received: %d.%d", msg->name.c_str(), msg->time.sec,
+             msg->time.nsec);
+}
+
+bool serviceCallback(server::serviceRequest &req, server::serviceResponse &) {
+    if (req.id == 0) {
+        sublist.insert(
+                std::pair<std::string, ros::Subscriber>(req.name,
+                                                        nhp->subscribe(req.name,
+                                                                       1000,
+                                                                       topicCallBack)));
```

Comment:
**[cpplint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> Lines should be <= 80 characters long  [whitespace/line_length] [2]